### ISSUE TITLE
CalCurse-CalDav: Various UnicodeDecodeErrors

### ISSUE THREAD

**Issue by bbenne10** [lfos/calcurse#19]
In testing the implementation of calcurse-caldav, I have stumbled upon some strange errors. I've got some data that cannot be decoded by the `utf-8` codec and it's raising `UnicodeDecodeError` in  `calcurse_export` where it bubbles to `push_object` and then to `push_objects` and `calcurse_import` where it bubbles to `pull_object` and then `pull_objects`, meaning all import/export operations fail because of one oddly formatted message).

I have changed `calcurse_{export,import}` to except `UnicodeDecodeError` and return None, which bubbles through `{push,pull}_object` to `{push,pull}_objects`, thus not raising an error. This might not be the proper solution though, so I was hoping to start discussion. If anyone is interested, I'll happily open a PR against this ticket to discuss implementation directly.

**Comment by lfos** [comment]
I think we should fix the underlying cause rather than ignoring the
error. What exactly is the problem here? Is the description stored in a
different encoding in the calcurse data file such that it cannot be
processed by calcurse-caldav which assumes UTF-8 encoding? Can you
temporarily work around the issue by switching your locale to something
UTF-8 based?

**Comment by bbenne10** [comment]
I'm away from the machine exhibiting these issues today and so can't test, but my locale is 'en_US.UTF-8' on that box so that's a no-go. I should also have mentioned that I was running with `--init=keep-remote`  and a presumably empty local calendar so these should all be events coming from the server. I did have a calcurse instance running at one point in the past, and thus should go back and check that there's no leakage between the two. I'll remove any existing calendar and start over again.

I agree that fixing the underlying issue is a more ideal solution, but I am currently rather uncertain about the root cause of those changes in encoding or of the actual correct encoding for these events. Admittedly I didn't try 'utf-16', but I did try a small handful of Windows' encodings (particularly `ISO-8859-1`, `ISO-8859-2`, and `Windows-1252`) to no avail. When I get into the office tomorrow, I'll try `utf-16` and report back.  Further, I'll see if there's some indication of encoding on the messages themselves (though I didn't find any on a cursory inspection when debugging this the first time).
